contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+                    deadline: block.timestamp,
+                    sqrtPriceX96: 0
+                })
+            );
+        } else {
+            (, _wethAmountSupplied, _oethbAmountSupplied) = positionManager
+                .increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+
+        /**
+         * First check we are in expected tick range
+         *
+         * We revert even though price being equal to the lower tick would still
+         * count being within lower tick for the purpose of Sugar.estimateAmount calls
+         */
+        if (
+            _currentPrice <= sqrtRatioX96TickLower ||
+            _currentPrice >= sqrtRatioX96TickHigher
+        ) {
+            revert OutsideExpectedTickRange(getCurrentTradingTick());
+        }
+
+        /**
+         * If estimateAmount1 call fails it could be due to _currentPrice being really
+         * close to a tick and amount1 too big to compute.
+         *
+         * If token addresses were reversed estimateAmount0 would be required here
+         */
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOethAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(0), // no need to pass pool address when current price is specified
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        // 18 decimal number expressed weth tick share
+        uint256 _wethSharePct = _normalizedWethAmount.divPrecisely(
+            _normalizedWethAmount + _correspondingOethAmount
+        );
+
+        if (
```

Comment:
⚪️ This is an if -> success -> else -> revert pattern. It should be written if-> failure

Reply:
good simplification 👍 Fix here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/c8c8bd7b3f9724356d20f60fc743850b2e1c8ff5